UOP1_Project/Assets/Scripts/Events/ScriptableObjects/UI/FadeChannelSO.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+[CreateAssetMenu(menuName = "Events/UI/Fade Channel")]
+public class FadeChannelSO : ScriptableObject
+{
+	public UnityAction<bool, float, Color> OnEventRaised;
+	/// <summary>
+	/// Generic fade function. Communicates with <seealso cref="FadeManager.cs"/>.
+	/// </summary>
+	/// <param name="fadeIn">If true, the rectangle fades in. If false, the rectangle fades out.</param>
+	/// <param name="duration">How long it takes to the image to fade in/out.</param>
+	/// <param name="color">Target color for the image to reach. Disregarded when fading out.</param>
+	public void Fade(bool fadeIn, float duration, Color color)
+	{
+		if (OnEventRaised != null)
+			OnEventRaised.Invoke(fadeIn, duration, color);
+	}
+
+	/// <summary>
+	/// Generic fade function. Communicates with <seealso cref="FadeManager.cs"/>.
+	/// </summary>
+	/// <param name="fadeIn">If true, the rectangle fades in. If false, the rectangle fades out.</param>
+	/// <param name="duration">How long it takes to the image to fade in/out.</param>
+	public void Fade(bool fadeIn, float duration)
+	{
+		if (OnEventRaised != null)
+			OnEventRaised.Invoke(fadeIn, duration, new Color(0, 0, 0, 1));
+	}
```

Comment:
You can give parameters a default value so that you don't have to write the same method multiple times.
Here an example:
```suggestion
	public void Fade(bool fadeIn, float duration, Color color = default(Color))
	{
		if (color == default(Color)) color = Color.black;
		if (OnEventRaised != null)
			OnEventRaised.Invoke(fadeIn, duration, color);
	}
```
This would work for the FadeIn-Methods as well

FYI: default(Color) == new Color(0, 0, 0, 0) and that's the only problem: If you want to fade in to or out from transparent but I don't see how that makes any sense so I would ignore that